web/blog/shelling-out-to-zeek-and-suricata/index.md

```diff
@@ -0,0 +1,179 @@
+bedrock for many security teams. But operationalizing these tools can take a
+good chunk of time.
+
+So we asked ourselves: **How can we make it super easy to work with Zeek and
+Suricata logs?**
+
+![Shell Operator](shell-operator.excalidraw.svg)
+
+<!--truncate-->
+
+## Purring like a Suricat
+
+In our [previous blog post](/blog/zeek-and-ye-shall-pipe/) we adapted Zeek to
+behave like a good ol' Unix tool, taking input via stdin and producing output
+via stdout. Turns out you can do the same fudgery with Suricata:[^1]
+
+[^1]: Suricata outputs [EVE
+JSON](https://suricata.readthedocs.io/en/latest/output/eve/eve-json-output.html)
+by default, which is equivalent to Zeek's streaming JSON output, with the
+difference being that Zeek's `_path` field is called `event_type` in Suricata
+logs.
+
+```bash title=suricatify
+#!/bin/sh
+suricata -r /dev/stdin \
+  --set outputs.1.eve-log.filename=/dev/stdout \
+  --set logging.outputs.0.console.enabled=no
+```
+
+Let's break this down:
+
+- The `--set` option take a `name=value` parameter that overrides the settings
+  in your `suricata.yaml` config file.
+- The key `outputs.1.eve-log.filename` refers to the `outputs` array, takes
+  element at index `1`, treats that as object and goes to the nested field
+  `eve-log.filename`. Setting `/dev/stdout` as filename makes Suricata write to
+  stdout.
+- We must set `logging.outputs.0.console.enabled` to `no` because Suricata
+  writes startup log messages to stdout. Since they are not valid JSON, we
+  would otherwise create an invalid JSON output stream.
+
+## User-defined Operators
+
+Now that we have both Zeek and Suricata at our fingertips, how can we work with
+their output more easily? This is where Tenzir comes into play—easy
+[pipelines](/language/pipelines) for security teams to acquire,
+[reshape](/user-guides/reshape-data), and route event data.
+
+Here are two examples that count the number of unique source IP addresses per
+destination IP address, on both Zeek and Suricata data:
+
+```bash
+# Zeek
+zcat pcap.gz | zeekify | tenzir \
+  'read zeek-json
+   | where #schema == "zeek.conn"
+   | summarize n=count_distinct(id.orig_h) by id.resp_h
+   | sort n desc'
+# Suricata
+zcat pcap.gz | suricatify | tenzir \
+  'read suricata
+   | where #schema == "suricata.flow"
+   | summarize n=count_distinct(src_ip) by dst_ip
+   | sort n desc'
+```
+
+It's a bit unwieldy to write such a command line that requires an external shell
+script to work. This is where [user-defined operators](/operators/user-defined)
+come into play. In combination with the [`shell`](/operators/sources/shell)
+operator, you can write a custom `zeek` and `suricata` operator and ditch the
+shell script:
+
+```yaml title="tenzir.yaml"
+tenzir:
+  operators:
+    zeek: >
+     shell "zeek -r - LogAscii::output_to_stdout=T
+            JSONStreaming::disable_default_logs=T
+            JSONStreaming::enable_log_rotation=F
+            json-streaming-logs"
+     | parse zeek-json
```

Comment:
Opening this conversation here to remind us of the `parse -> read` change, which is scheduled to happen before this post is published.

Reply:
Since that PR was just merged we need to replace parse with read in this blog post before merging.